expression/aggregation/descriptor.go

```diff
@@ -110,39 +110,86 @@ func (a *AggFuncDesc) typeInfer(ctx sessionctx.Context) {
+// Query: `select t.a as `t.a`,  count(95), sum(95), avg(95), bit_or(95), bit_and(95), bit_or(95), max(95), min(95), s.a as `s.a`, avg(95) from t left join s on t.a = s.a;`
+// +------+-----------+---------+---------+------------+-------------+------------+---------+---------+------+----------+
+// | t.a  | count(95) | sum(95) | avg(95) | bit_or(95) | bit_and(95) | bit_or(95) | max(95) | min(95) | s.a  | avg(s.a) |
+// +------+-----------+---------+---------+------------+-------------+------------+---------+---------+------+----------+
+// |    1 |         1 |      95 | 95.0000 |         95 |          95 |         95 |      95 |      95 | NULL |     NULL |
+// +------+-----------+---------+---------+------------+-------------+------------+---------+---------+------+----------+
+func (a *AggFuncDesc) EvalNullValueInOuterJoin(ctx sessionctx.Context, schema *expression.Schema) (types.Datum, bool) {
 	switch a.Name {
 	case ast.AggFuncCount:
-		return a.calculateDefaultValue4Count(ctx, schema)
+		return a.evalNullValueInOuterJoin4Count(ctx, schema)
 	case ast.AggFuncSum, ast.AggFuncMax, ast.AggFuncMin,
-		ast.AggFuncFirstRow, ast.AggFuncAvg, ast.AggFuncGroupConcat:
-		return a.calculateDefaultValue4Sum(ctx, schema)
+		ast.AggFuncFirstRow:
+		return a.evalNullValueInOuterJoin4Sum(ctx, schema)
+	case ast.AggFuncAvg, ast.AggFuncGroupConcat:
+		return types.Datum{}, false
 	case ast.AggFuncBitAnd:
-		return a.calculateDefaultValue4BitAnd(ctx, schema)
+		return a.evalNullValueInOuterJoin4BitAnd(ctx, schema)
 	case ast.AggFuncBitOr, ast.AggFuncBitXor:
-		return a.calculateDefaultValue4BitOr(ctx, schema)
+		return a.evalNullValueInOuterJoin4BitOr(ctx, schema)
 	default:
 		panic("unsupported agg function")
 	}
 }
 
+// GetDefaultValue gets the default value when the aggregation function's input is null.
+// According to MySQL, default values of the aggregation function are listed as follows:
+// e.g.
+// Table t which is empty:
+// +-------+---------+---------+
+// | Table | Field   | Type    |
+// +-------+---------+---------+
+// | t     | a       | int(11) |
+// +-------+---------+---------+
+//
+// Query: `select a, avg(a), sum(a), count(a), bit_xor(a), bit_or(a), bit_and(a), max(a), min(a), group_concat(a) from t;`
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+// | a    | avg(a) | sum(a) | count(a) | bit_xor(a) | bit_or(a) | bit_and(a)           | max(a) | min(a) | group_concat(a) |
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+// | NULL |   NULL |   NULL |        0 |          0 |         0 | 18446744073709551615 |   NULL |   NULL | NULL            |
+// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+func (a *AggFuncDesc) GetDefaultValue() (v types.Datum) {
```

Comment:
Can this function be moved to the `Aggregation` interface, it seems like belonging to the execution layer.